src/services/prepareSubmissionData.ts

```diff
@@ -3,11 +3,16 @@ import { NewDraftSubmission } from '../types/submissions'
 import { executePendingQueueAttachmentProgressListeners } from './pending-queue'
 import uploadAttachment from './uploadAttachment'
 
-export default async function prepareSubmissionData({
-  definition,
-  submission,
-}: NewDraftSubmission): Promise<NewDraftSubmission['submission']> {
-  return uploadAttachments(definition.id, definition.elements, submission)
+export default async function prepareSubmissionData(
+  { definition, submission }: NewDraftSubmission,
+  continueWhilstAttachmentsAreUploading?: boolean,
+): Promise<NewDraftSubmission['submission']> {
+  return uploadAttachments(
```

Comment:
Can we just do this check here once? Do we need to iterate over the submission?

Reply:
yes, my bad, wrongly thought it was doing over stuff with the submission